creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -913,15 +920,28 @@ class SnapbackSM {
       'BEGIN processStateMachineOperation()',
       {
         currentModuloSlice: this.currentModuloSlice,
-        moduloBase: this.moduloBase
+        moduloBase: this.moduloBase,
+        lastProcessedUserId: this.lastProcessedUserId
       }
     )
 
+    let nodeUsers = []
+    // New DN versions support pagination, so we fall back to modulo slicing for old versions
+    // TODO: Remove modulo supports once all DNs update to include https://github.com/AudiusProject/audius-protocol/pull/3071
+    let useModulo = false
     try {
-      let nodeUsers
       try {
-        nodeUsers = await this.peerSetManager.getNodeUsers()
-        nodeUsers = this.sliceUsers(nodeUsers)
+        nodeUsers = await this.peerSetManager.getNodeUsers(
+          this.lastProcessedUserId,
+          this.usersPerJob
+        )
+
+        // Backwards compatibility -- if DN hasn't updated to support pagination,
+        // then we have to manually paginate the full set of users
+        if (nodeUsers.length > this.usersPerJob) {
+          useModulo = true
+          nodeUsers = this.sliceUsers(nodeUsers)
+        }
```

Comment:
why is it that if this condition is true that we have to rely on the modulo logic?

Reply:
this condition being true means DN returned all users, which means it ignored the `max_users` param we passed. it only ignores that param if it's not updated with the pagination logic